includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -62,6 +65,77 @@ public function maybe_init_subscriptions() {
 		add_action( 'template_redirect', [ $this, 'restore_order_pay_var' ], 101 );
 	}
 
+	/**
+	 * Add the necessary information to payment intents for subscriptions to allow Stripe to create
+	 * mandates for 3DS payments in India. It's ok to apply this across the board; Stripe will
+	 * take care of handling any authorizations.
+	 *
+	 * @param Array    $request          The HTTP request that will be sent to Stripe to create the payment intent.
+	 * @param WC_Order $order            The renewal order.
+	 * @param Array    $prepared_source  The source object.
+	 */
+	public function add_subscription_information_to_intent( $request, $order, $prepared_source ) {
+		// Just in case the order doesn't contain a subscription we return the base request.
+		if ( ! $this->has_subscription( $order->get_id() ) ) {
+			return $request;
+		}
+
+		// TODO: maybe this isn't necessary since this function should really only be called
+		//       when creating the intent? It's called in process_subscription_payment though
+		//       so it's probably needed here too?
+		// If we've already created a mandate for this order; use that.
+		$mandate = $order->get_meta( '_stripe_mandate_id', true );
+		if ( isset( $mandate ) && '' !== $mandate ) {
+			$request['mandate'] = $mandate;
+			return $request;
+		}
+
+		// TODO: what happens if there's more than one renewal order?
+		$renewals = wcs_get_subscriptions_for_renewal_order( $order->get_id() );
+		if ( 1 === count( $renewals ) ) {
+			$renewal_order   = reset( $renewals );
+			$parent_order_id = $renewal_order->get_parent_id();
+			$parent_order    = wc_get_order( $parent_order_id );
+
+			$mandate = $parent_order->get_meta( '_stripe_mandate_id', true );
+			if ( isset( $mandate ) && '' !== $mandate ) {
+				$request['mandate'] = $mandate;
+				return $request;
+			}
+		}
```

Comment:
I think in this case it's fine that there's more than one order; they'd all presumably have the same mandate, so this is probably fine as is 🤔